src/js/polyfills/webvtt.js

```diff
@@ -457,15 +458,13 @@ function CueStyleBox(window, cue) {
     switch (cue.align) {
         case 'start':
         case 'left':
+        case 'end':
+        case 'right':
             textPos = cue.position;
             break;
         case 'middle':
         case 'center':
-            textPos = (cue.position === 'auto' ? 50 : cue.position) - (cue.size / 2);
-            break;
-        case 'end':
-        case 'right':
-            textPos = cue.position - cue.size;
+            textPos = (cue.position === 'auto' ? 0 : cue.position);
```

Comment:
this assumes that `align:end` with `position:0%` expects the text to be right aligned but present at the beginning of the captions container, if this is not correct - can update

Reply:
just checked the vtt spec - looks like this needs to be reverted